linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -881,17 +885,17 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
           foldAsInstanceOf(texpr, tpe)(cont)
         }
 
-      case Closure(arrow, captureParams, params, body, captureValues) =>
+      case Closure(arrow, captureParams, params, restParam, body, captureValues) =>
         pretransformExprs(captureValues) { tcaptureValues =>
           def default(): TailRec[Tree] = {
             val newClosure = transformClosureCommon(arrow, captureParams,
-                params, body, tcaptureValues.map(finishTransformExpr))
+                params, restParam, body, tcaptureValues.map(finishTransformExpr))
             cont(PreTransTree(
                 newClosure,
                 RefinedType(AnyType, isExact = false, isNullable = false)))
           }
 
-          if (!arrow || params.exists(_.rest)) {
+          if (!arrow || restParam.nonEmpty) {
```

Comment:
`restParam.isDefined` is perhaps bit more canonical, and even clearer in this case?